### PR TITLE
Added unit test for Webserver.cc

### PR DESCRIPTION
**Pull request by tyhchoi**
Did not test this yet

### CODE REVIEW

File: Webserver_test.cc [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#include <cstdlib>
+#include <iostream>
+#include <boost/asio.hpp>
+#include "gtest/gtest.h"
+#include "webserver.cc"
```
Comment (llam15) [inline Webserver_test.cc]: #include "Webserver.h"

File: Webserver_test.cc [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+TEST(WebserverTest, run_serverTest) {
+	  boost::asio::io_service io_service;
+	  Webserver server;
+	  server.run_server(io_service, 8080);
+	  tcp::resolver resolver (io_service);
```
Comment (llam15) [inline Webserver_test.cc]: You might need to use the namespace boost::asio::ip::tcp

File: Webserver_test.cc [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	  tcp::resolver::query query(localhost:8080, "asdf");
+    tcp::resolver::iterator endpoint_iterator = resolver.resolve(query);
```
Comment (llam15) [inline Webserver_test.cc:22]: What's happening with the indents lmao

File: Webserver_test.cc [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+#include <boost/asio.hpp>
+#include "gtest/gtest.h"
+#include "webserver.cc"
+
+TEST(WebserverTest, run_serverTest) {
+	  boost::asio::io_service io_service;
+	  Webserver server;
+	  server.run_server(io_service, 8080);
+	  tcp::resolver resolver (io_service);
+	  tcp::resolver::query query(localhost:8080, "asdf");
```
Comment (llam15) [inline Webserver_test.cc]: tcp::resolver::query query(tcp::v4(), localhost, 8080);

File: Webserver_test.cc [hunk 5]
```diff
@@ -0,0 +1,34 @@
+#include <cstdlib>
+#include <iostream>
+#include <boost/asio.hpp>
+#include "gtest/gtest.h"
+#include "webserver.cc"
+
+TEST(WebserverTest, run_serverTest) {
+	  boost::asio::io_service io_service;
+	  Webserver server;
+	  server.run_server(io_service, 8080);
```
Comment (llam15) [inline Webserver_test.cc]: We changed the implementation of run_server. You will need to run server.parse_config(config_file) then server.run_server(io_service)

File: Webserver_test.cc [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+    tcp::socket socket(io_service);
+    boost::asio::connect(socket, endpoint_iterator);
+
+    boost::asio::streambuf request;
+    std::ostream request_stream(&request);
+    request_stream << "asdfasdf";
+
+    // Send the request.
+    boost::asio::write(socket, request);
```
Comment (llam15) [inline Webserver_test.cc:32]: I think you have to pass in request.data()?

**Review by llam15 — CHANGES_REQUESTED** [review]

File: Webserver_test.cc (hunk 4 above)
Comment (tyhchoi) [reply Webserver_test.cc]: i found out its supposed to be query("localhost", "8080");

File: Webserver_test.cc (hunk 6 above)
Comment (tyhchoi) [reply Webserver_test.cc:32]: i think the first argument is the socket and the next is the buffer

**Pull request closed without merging**